pkg/db/driver_mysql.go

```diff
@@ -24,25 +23,21 @@ func NewMysqlDriverFactory() DriverFactory {
 type mysqlDriverFactory struct{}
 
 func (m *mysqlDriverFactory) GetDSN(settings Settings) string {
-	dsn := url.URL{
-		User: url.UserPassword(settings.Uri.User, settings.Uri.Password),
-		Host: fmt.Sprintf("tcp(%s:%d)", settings.Uri.Host, settings.Uri.Port),
-		Path: settings.Uri.Database,
-	}
-
-	qry := dsn.Query()
-	qry.Set("multiStatements", "true")
-	qry.Set("parseTime", strconv.FormatBool(settings.ParseTime))
-	qry.Set("charset", "utf8mb4")
-	dsn.RawQuery = qry.Encode()
-
-	uri := dsn.String()
-
-	return uri[2:]
+	cfg := mysql.NewConfig()
+	cfg.User = settings.Uri.User
+	cfg.Passwd = settings.Uri.Password
+	cfg.Net = "tcp"
+	cfg.Addr = net.JoinHostPort(settings.Uri.Host, strconv.Itoa(settings.Uri.Port))
+	cfg.DBName = settings.Uri.Database
+	cfg.MultiStatements = true
+	cfg.ParseTime = true
+	cfg.Collation = "utf8mb4"
+
+	return cfg.FormatDSN()
 }
 
 func (m *mysqlDriverFactory) GetMigrationDriver(db *sql.DB, database string, migrationsTable string) (database.Driver, error) {
-	return mysql.WithInstance(db, &mysql.Config{
+	return migrateMysql.WithInstance(db, &migrateMysql.Config{
```

Comment:
interesting, i didn't realize we are still using the golang migrate library here when i reviewed the goose db changes. Should this change as well?

Reply:
In fact this whole GetMigrationDriver func does only seem to be used here: [pkg/db/migrations_golang_migrate.go](https://github.com/justtrackio/gosoline/blob/c6717629ecc235748824b1f4735e18ff368605b4/pkg/db/migrations_golang_migrate.go#L31)

Do we want to deprecate it or re/move it immediately?

Reply:
ya, could be that this could be improved.. but is a little bit out of scope for me now..